### PR TITLE
Added bead sort

### PR DESCRIPTION
**Pull request by mateuszz0000**
### **Describe your change:**
Added bead sort aka gravity sort

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/bead_sort.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+""" https://en.wikipedia.org/wiki/Bead_sort """
+
+
+def bead_sort(sequence: list) -> list:
+    """
+    >>> bead_sort([6,11, 12, 4, 1, 5])
```
Comment (cclauss) [inline sorts/bead_sort.py:6]:
```suggestion
    >>> bead_sort([6, 11, 12, 4, 1, 5])
```

Comment (cclauss) [reply sorts/bead_sort.py]: Please test [1, .9, 0.0, 0, -1, -.9] and "Hello World!"

Comment (mateuszz0000) [reply sorts/bead_sort.py]: I am wondering on "Hello world" ordering. Theory says that algorithm should be use for positive integers. This implementation works with `[1, .9, 0.0, 0, -1, -.9]` but should it support characters too?

Comment (mateuszz0000) [reply sorts/bead_sort.py]: After a few moments, I decide to write two checks for it. I don't want to argue with the theory! :smile:  @cclauss maybe you have a better idea for this?

File: sorts/bead_sort.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    >>> bead_sort("Hello world")
+    Traceback (most recent call last):
+    ...
+    TypeError: Sequence must be list of positive integers
+    """
+    if isinstance(sequence, str):
+        raise TypeError("Sequence must be list of positive integers")
+
+    if min(sequence) < 0:
+        raise TypeError("Sequence must be list of positive integers")
```
Comment (cclauss) [inline sorts/bead_sort.py:32]: Why not sort these things?  They should work just fine.

**Pull request merged**